### PR TITLE
SargisVa homework6 2-12-2021

### PR DESCRIPTION
**Pull request by SargisV**

### CODE REVIEW

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask596.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			if (a >= c) {
+				max = a;
+			} else {
+				max = c;
+			}
+		} else if (b <= c) {
+			max = c;
+		} else {
+			max = b;
+		}
```
Comment (sirak92) [inline homeworks/sargis_vardanyan/homework6/methods/MethodsTask596.java:38]: it's correct but not a good solution
The simplest solution would be:
max = a;
if (max < b)
  max = b;
if (max < c)
  max = c;
I remember we have the same conversation earlier as well :)

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask604.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+			if (y < z) {
+				min = y;
+			} else {
+				min = z;
+			}
+		} else if (x < z) {
+			min = x;
+		} else {
+			min = z;
+		}
```
Comment (sirak92) [inline homeworks/sargis_vardanyan/homework6/methods/MethodsTask604.java:46]: same here

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask616.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		int countOfPositives = 0;
+		
+		for (int i = 0; i < array.length; i++) {
+			if (array[i] >= 0) {
+				sumOfPositives += array[i];
+				countOfPositives++;
+			}
+		}
+		
+		return (double) (sumOfPositives / countOfPositives);
```
Comment (sirak92) [inline homeworks/sargis_vardanyan/homework6/methods/MethodsTask616.java:42]: what the (double) is doing in this case? isn't it redundant?

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask620.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		int a = scanner.nextInt();
+		System.out.println("Enter b: ");
+		int b = scanner.nextInt();
+		
+		System.out.println("sum: " + sumOf(a, b, array));
+		
+	
+	}
+	
+	//returns sam of array's elements that are in range of [a, b].
```
Comment (sirak92) [inline homeworks/sargis_vardanyan/homework6/methods/MethodsTask620.java:36]: actually this kind of comments are mainly written with java doc comments

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask625.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		
+		return max;
+	}
+	
+	//returns multiplication of array's maximum elements indexes.
+	static int multOfIndexes(int[] array) {
+		int mult = 1;
+		
+		for (int i = 0; i < array.length; i++) {
+			if (array[i] == getMax(array)) {
```
Comment (sirak92) [inline homeworks/sargis_vardanyan/homework6/methods/MethodsTask625.java:48]: is the max changed during the iteration? no, then you need to call the function outside of for() to make the code efficience

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask627.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+	//initializing and returning new array which elements are array[n]'s negative members indexes.
+	static int[] getNewArray(int[] array) {
+		
+		int[] newArray = new int[countOfNegatives(array)];
+		int index = 0;
+		
+		for (int i = 0; i < array.length; i++) {
+			if (array[i] < 0) {
+				newArray[index] = i;
+				index++;
```
Comment (sirak92) [inline homeworks/sargis_vardanyan/homework6/methods/MethodsTask627.java:57]: newArray[index++] = i

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask596.java (hunk 1 above)
Comment (SargisV) [reply homeworks/sargis_vardanyan/homework6/methods/MethodsTask596.java:38]: yeah

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask604.java (hunk 2 above)
Comment (SargisV) [reply homeworks/sargis_vardanyan/homework6/methods/MethodsTask604.java:46]: just remember

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask616.java (hunk 3 above)
Comment (SargisV) [reply homeworks/sargis_vardanyan/homework6/methods/MethodsTask616.java:42]: sumOfPositives and countOfPositives are announced int

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask620.java (hunk 4 above)
Comment (SargisV) [reply homeworks/sargis_vardanyan/homework6/methods/MethodsTask620.java:36]: ok

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask625.java (hunk 5 above)
Comment (SargisV) [reply homeworks/sargis_vardanyan/homework6/methods/MethodsTask625.java:48]: something like this	
int mult = 1;
		int max = getMax(array));
		
		for (int i = 0; i < array.length; i++) {
			if (array[i] == max) {
				mult *= i;
			}
		}

File: homeworks/sargis_vardanyan/homework6/methods/MethodsTask627.java (hunk 6 above)
Comment (SargisV) [reply homeworks/sargis_vardanyan/homework6/methods/MethodsTask627.java:57]: right

**Pull request merged**